test/e2e/density.go

```diff
@@ -231,9 +235,38 @@ var _ = Describe("Density", func() {
 		}
 		itArg := testArg
 		It(name, func() {
+			uuid := string(util.NewUUID())
 			totalPods := itArg.podsPerMinion * minionCount
-			RCName = "my-hostname-density" + strconv.Itoa(totalPods) + "-" + string(util.NewUUID())
+			nameStr := strconv.Itoa(totalPods) + "-" + uuid
+			ns = "e2e-density" + nameStr
+			RCName = "my-hostname-density" + nameStr
 			RunRC(c, RCName, ns, "gcr.io/google_containers/pause:go", totalPods)
+			By("waiting for all events to be recorded")
+			last := -1
+			current := 0
+			var events *api.EventList
+			for last < current {
+				e, err := c.Events(ns).List(
+					labels.Everything(),
+					fields.Set{
+						"involvedObject.namespace": ns,
+					}.AsSelector(),
+				)
+				expectNoError(err)
+				last = current
```

Comment:
If the concern is the efficiency of the test, then I would rather get the change in now and enhance it later.  Ensuring the condition is tested is more important than ensuring it is done in the most efficient way imo.  I couldn't say that any other e2e test is performing its functions in the most efficient way possible.

My concern has been bounding the event collection loop, and I've done that in a change I can push.